ddtrace/propagation/http.py

```diff
@@ -839,7 +839,8 @@ def _resolve_contexts(contexts, styles_w_ctx, normalized_headers):
         links = []
         for context in contexts[1:]:
             style_w_ctx = styles_w_ctx[contexts.index(context)]
-            if context.trace_id != primary_context.trace_id:
+            # encoding expects at least trace_id and span_id
+            if context.span_id and context.trace_id and context.trace_id != primary_context.trace_id:
                 links.append(
                     SpanLink(
```

Comment:
Can we update the typing on the fields in the SpanLink class? We should make sure trace_id and span_id are always integers. It would be nice to also add a validator that prevents the ids  from being zero too. 

This can be done in a follow up. It's not a blocker

Reply:
I'll do it as follow up.